packages/screens/LaunchpadERC20/LaunchpadERC20Sales/LaunchpadERC20CreateSaleForm.tsx

```diff
@@ -0,0 +1,200 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import React, { useEffect, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { TouchableOpacity, View } from "react-native";
+
+import { LaunchpadERC20CreateSaleFooter } from "./LaunchpadERC20CreateSaleFooter";
+import { useCreateSaleState } from "../hooks/useCreateSale";
+import { TCreateSaleForm, zodCreateSaleForm } from "../utils/forms";
+
+import { BrandText } from "@/components/BrandText";
+import ToggleButton from "@/components/buttons/ToggleButton";
+import { CsvTextRowsInput } from "@/components/inputs/CsvTextRowsInput";
+import { DateTimeInput } from "@/components/inputs/DateTimeInput";
+import { TextInputCustom } from "@/components/inputs/TextInputCustom";
+import { SpacerColumn } from "@/components/spacer";
+import useSelectedWallet from "@/hooks/useSelectedWallet";
+import { computeMerkleRoot } from "@/utils/merkle";
+import { neutral77 } from "@/utils/style/colors";
+import { fontSemibold14, fontSemibold20 } from "@/utils/style/fonts";
+
+// TODO: ADD A WAY TO JUST DROP A CSV FILE OR ENTER A LIST OF ADDR AND IT WILL COMPUTE THE MERKLE ROOT FOR YOU
+// TODO: ADD A DATE PICKER FOR THE START TIMESTAMP AND END TIMESTAMP
+export const CreateSaleForm: React.FC = () => {
+  const [whitelistedAddresses, setWhitelistedAddresses] = useState<string[]>();
+  const {
+    actions: { goNextStep, setSale },
+    createSaleForm: formData,
+  } = useCreateSaleState();
+  const selectedWallet = useSelectedWallet();
+  const caller = selectedWallet?.address;
+  const { handleSubmit, setValue, watch, control, getFieldState } =
+    useForm<TCreateSaleForm>({
+      resolver: zodResolver(zodCreateSaleForm),
+      defaultValues: formData,
+      mode: "all",
+    });
+
+  useEffect(() => {
+    if (!caller) {
+      // TODO: would be better to not allow this corner case, aka do something smarter when no wallet is connected
+      return;
+    }
+    setValue("caller", caller);
+  }, [setValue, caller]);
+
+  if (!caller) {
+    return <BrandText>Connect a wallet to create a sale</BrandText>;
+  }
+
+  const values = watch();
+  const startTimestamp = watch("startTimestamp");
+  const endTimestamp = watch("endTimestamp");
+  return (
+    <View style={{ width: "100%", maxWidth: 480, margin: "auto" }}>
+      <BrandText style={fontSemibold20}>Sale Informations</BrandText>
+
+      <SpacerColumn size={1} />
+
+      <BrandText style={[fontSemibold14, { color: neutral77 }]}>
+        Information about your Sale
+      </BrandText>
+
+      <SpacerColumn size={2.5} />
+
+      <TextInputCustom<TCreateSaleForm>
+        label="Token Name (The token have to be owned by the caller)"
+        name="tokenName"
+        fullWidth
+        placeholder="Type the token name here..."
+        variant="labelOutside"
+        control={control}
+        rules={{ required: true }}
```

Comment:
Could you make the inputs height as `40` ?
On `TextInputCustom`, you can add `height={40}`
Sorry, we have to refacto the default inputs style ^^'

```suggestion
        rules={{ required: true }}
        height={40}
```